solr/core/src/java/org/apache/solr/core/SolrConfig.java

```diff
@@ -224,6 +224,13 @@ private SolrConfig(SolrResourceLoader loader, String name, boolean isConfigsetTr
     queryResultWindowSize = Math.max(1, getInt("query/queryResultWindowSize", 1));
     queryResultMaxDocsCached = getInt("query/queryResultMaxDocsCached", Integer.MAX_VALUE);
     enableLazyFieldLoading = getBool("query/enableLazyFieldLoading", false);
+
+    useCircuitBreakers = getBool("query/useCircuitBreakers", false);
+    memoryCircuitBreakerThreshold = getInt("query/memoryCircuitBreakerThreshold", 100);
+
+    if (memoryCircuitBreakerThreshold > 100 || memoryCircuitBreakerThreshold < 0) {
```

Comment:
We only want to do this validation if circuitBreakers are enabled, right?

Reply:
I was being overtly cautious here -- it seems wrong if the parameter is incorrectly specified regardless of whether it is used or not?

Reply:
Perhaps you want to change this to early saying "Circuit breaker disabled, but incorrectly configured" or <current message> conditionally?

Reply:
I have changed the check to be only performed when circuit breakers are enabled.